### ISSUE TITLE
asExpression unwanted error

### ISSUE THREAD

**Issue by Fandekasp** [eslint/typescript-eslint-parser#563]
Related to #25 

**What version of TypeScript are you using?**
3.1.6

**What version of `typescript-eslint-parser` are you using?**
21.0.1

**What code were you trying to parse?**

```ts
interface CustomWindow extends Window {
  gtmDataLayer?: any;
}

…
      (window as CustomWindow).gtmDataLayer.push({
        event: "Registration started",
        provider,
      });

```

**What did you expect to happen?**
No error shown in vim, similarly to tsc reportt
```
$ yarn type-check
yarn run v1.12.3
$ tsc --project tsconfig.json
Done in 3.81s.
```

**What happened?**
![](https://i.imgur.com/mZXG7xH.png)

I spent hours thinking that the code above was incorrect because of that error before realizing that tsc didn't raise errors on this code.

The problem is that my git hook fails with the eslint error, and that's not an error I can eslint-disable..

**Comment by Fandekasp** [comment]
Sorry about that. We're not using the typescript-eslint-parser but babel-eslint
